src/dep-id/src/index.ts

```diff
@@ -96,11 +96,23 @@ export const joinDepIDTuple = (list: DepIDTuple): DepID => {
 }
 
 // allow @, but otherwise, escape everything urls do
-const encode = (s: string): string =>
-  encodeURIComponent(s)
-    .replaceAll('%40', '@')
-    .replaceAll('%2f', '§')
-    .replaceAll('%2F', '§')
+const encode = (s?: string) =>
+  s ?
+    encodeURIComponent(s)
+      .replaceAll('%40', '@')
+      .replaceAll('%2f', '§')
+      .replaceAll('%2F', '§')
+  : s
+
+const decode = (s?: string) =>
+  s ?
+    decodeURIComponent(
+      s
+        .replaceAll('@', '%40')
+        .replaceAll('§', '%2f')
+        .replaceAll('§', '%2F'),
```

Comment:
The last two replacements are both for the same character right? So for decoding we should just pick which one of `%2f` and `%2F` we would prefer be in the result.

Reply:
good catch!